buildSrc/src/main/java/org/apache/lucene/gradle/WrapperDownloader.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.gradle;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.math.BigInteger;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+
+public class WrapperDownloader {
+    public static void main(String[] args) throws Exception {
+        if (args.length != 3) {
+            System.err.println("Usage: java WrapperDownloader.java <version> <destination> <checksum>");
+            System.exit(1);
+        }
+
+        File jar = new File(args[1]);
+        if (!jar.exists()) {
+            final URL url = new URL("https://github.com/gradle/gradle/raw/v" + args[0] + "/gradle/wrapper/gradle-wrapper.jar");
+
+            // As of v6.0.1 the wrapper is approximately 60K
+            // Can increase this is gradle wrapper ever goes beyond 100K, but keep a safety check
+            final int maxSize = 100 * 1024;
+
+            File temp = Files.createTempFile("gradle-wrapper", null).toFile();
+            try (ReadableByteChannel in = Channels.newChannel(url.openStream());
+                 FileOutputStream out = new FileOutputStream(temp)) {
+                out.getChannel().transferFrom(in, 0, maxSize);
+            }
+
+            try (FileInputStream fis = new FileInputStream(temp)) {
+                MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
+
+                // Convert binary digest to hex checksum value
+                // This will strip leading zeroes, deal with that when we need to
+                String sha256sum = new BigInteger(1, sha256.digest(fis.readAllBytes())).toString(16);
+
+                if (args[2].equals(sha256sum)) {
+                    if (!temp.renameTo(jar)) {
+                        System.err.println("Failed renaming wrapper jar, leaving download at " + temp);
```

Comment:
Always cleanup the temp file. What's the use of leaving it?

Reply:
The user might be able to manually move it into place since we've already validated the checksum. I'll update the error.